packages/vm/lib/state/stateManager.ts

```diff
@@ -542,6 +575,71 @@ export default class DefaultStateManager implements StateManager {
     return false
   }
 
+  /** EIP-2929 logic
+   * This should only be called from within the EVM
+   */
+
+  /**
+   * Returns true if the address is warm in the current context
+   * @param address - The address (as a Buffer) to check
+   */
+  isWarmAddress(address: Buffer): boolean {
+    for (let i = this._accessedStorage.length; i >= 0; i--) {
+      const currentMap = this._accessedStorage[this._accessedStorage.length - 1]
+      if (currentMap.has(address.toString('hex'))) {
```

Comment:
If I read the code correctly there seems to be no reason for this reverse order traversing (?) and this could then be simplified to `for (const currentMap of this._accessedStorage)`.

Reply:
I did this in reverse order since this is an optimization. Imagine that we are rather deep in our call stack, say 10 levels deep. If we now access a storage slot on the current address (assuming  that this is the first time we are at this address) then we actually mark the slot as warm in the end of the array. If we traverse the array in normal order, we first have to check the 9 other maps before figuring out that the 10th map (at current address) has marked it as warm. (It is a small optimization)